### DISCUSSION TITLE
Typescript select and usestate errors

### DISCUSSION THREAD

**Discussion by Tarbez** [supabase/supabase#9879] [Questions]
GM Fam, 

I'm very new to typescript and it seems like my code has a lot of errors in it. 

I managed to get the data into the console but it seems like I still have errors in the data

My code here: 
```
import { useEffect, useState } from "react";
import { supabase } from "../utils/supabase";

export default function Home() {
  const [name, setName] = useState<string>("");

  useEffect(() => {
    const CheckSpaceContentAllowance = async () => {
      interface Data {
        paper_title: string;
        paper_description: string;
      }

      const { data, error } = await supabase.from<Data>("papers").select(`
    paper_title,
    paper_description
  `);
      setName(data);
      console.log(data);
    };
    CheckSpaceContentAllowance();


    return () => {};
  }, []);

  return <div>Check Console</div>;
}

```
Update: 
Not sure why "data" is red. 
Despite the error within VScode, the site still does work and does bring the data as expected. I just don't get why it is underlined with red. 

![Screenshot 2022-10-27 at 2 18 09](https://user-images.githubusercontent.com/33286159/198156484-b003fa80-a8fe-467c-b570-691902dbe907.png)

I would appreciate some help. Thank you.

**Comment by singhBinary** [comment]
Maybe this can help https://supabase.com/docs/guides/api/rest/generating-types
